### PR TITLE
Campaign Shipment Form

### PR DESCRIPTION
**Pull request by aaronschachter**
Adds new helper variables to implement a Shipment Form on a campaign.

Creates Shipment Form modal on a campaign based on helper variable config.

Closes https://jira.dosomething.org/browse/DS-452.  Creates Shipments properly for non-shirt Shipment Items (Shirts need additional form elements for style and size, there are `@todo` comments for this, if we ever need to do that)

screen shot 2014-10-03 at 3 08 45 pm

### CODE REVIEW

**Comment by aaronschachter** [comment]
@sergii-tkachenko Can you review?

File: lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module [hunk 1]
```diff
@@ -245,12 +246,15 @@ function dosomething_shipment_get_shipment_id_by_entity($type, $id) {
...
  * Returns array of valid values for a Shipment's item / additional_item.
  */
-function dosomething_shipment_get_item_options() {
-  return array(
-    'banner' => t('Banner'),
-    'shirt' => t('Shirt'),
-    'thumb_socks' => t('Thumb Socks'),
-  );
+function dosomething_shipment_get_item_options($allow_empty = FALSE) {
+  $options = array();
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module:250]: Maybe you better use [`form_select_options`](https://api.drupal.org/api/drupal/includes%21form.inc/function/form_select_options/7)?

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module:250]: Any reason in particular?  We use similar callbacks in other modules that use Form API...

Comment (sergiitk) [inline lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module:250]: Actually, you're right. I just didn't like line 252. Will describe it below.

File: lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module [hunk 2]
```diff
@@ -245,12 +246,15 @@ function dosomething_shipment_get_shipment_id_by_entity($type, $id) {
...
+  if ($allow_empty) {
+    $options[''] = ' -- ';
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module]: Can we use [`#empty_option`](https://api.drupal.org/api/drupal/developer!topics!forms_api_reference.html/7#empty_option) instead?

File: lib/modules/dosomething/dosomething_shipment/dosomething_shipment.forms.inc [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  dosomething_user_save_address_values($values);
+
+  $shipment = entity_create('shipment', array(
+    'uid' => $user->uid,
+    'created' => REQUEST_TIME,
+    'entity_type' => $values['entity_type'],
+    'entity_id' => $values['entity_id'],
+    'item' => $values['shipment_item'],
+  ));
+  //@todo: If Shipment Item is shirt, add option / additional item/option values.
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_shipment/dosomething_shipment.forms.inc]: Take a look at Drupal Doxygen standarts for [inline todos](https://www.drupal.org/node/1354#inline).

File: lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module (hunk 2 above)
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_shipment/dosomething_shipment.module]: noted, will make this change

**Comment by sergiitk** [comment]
Looks :+1:, just few minor suggestions.

**Pull request merged**